src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysesAjaxController.java

```diff
@@ -204,11 +208,9 @@ private AnalysisModel createAnalysisModel(AnalysisSubmission submission, Locale
 		String stateString = messageSource.getMessage("analysis.state." + analysisState.toString(), null, locale);
 		AnalysisStateModel state = new AnalysisStateModel(stateString, analysisState.toString());
 		String workflow = messageSource.getMessage("workflow." + workflowType + ".title", null, workflowType, locale);
-		Long duration = 0L;
-		if (analysisState.equals(AnalysisState.COMPLETED)) {
-			duration = DateUtilities.getDurationInMilliseconds(submission.getCreatedDate(), submission.getAnalysis()
-					.getCreatedDate());
-		}
+
+		AnalysisAuditing analysisAudit = new AnalysisAuditing(entityManagerFactory);
```

Comment:
Why not make the `AnalysisAuditing` a component and have the `EntityManagerFactory` autowired into it?

Reply:
Removed Entity Manager/Audit Reader and annotated the class as a component in 6766c8ecf35d0d185900b8e1958aca1fa9d2d409